### PR TITLE
Pushforward feature from fe param search

### LINKED ISSUE #1740: Run pushforward trick experiments
### Describe the task. It can be a feature, documentation, etc.

The idea is to apply loss function on the last forecast step rather applying to all intermediate steps
The experiments will be conducted using both [fe-param-search](https://github.com/ecmwf/WeatherGenerator/tree/mk/develop/fe-param-search)  branch and `develop` branch.

### Hedgedoc URL, if you are keeping notes, plots, logs in hedgedoc.

_No response_

### Area

- [ ] datasets, data readers, data preparation and transfer
- [ ] model
- [x] science
- [ ] infrastructure and engineering
- [ ] evaluation, export and visualization
- [ ] documentation

### PR DESCRIPTION
**Pull request by ankitpatnala**
## Description

It takes around push-forward experiments where the loss is only calculated on 1 step rather than applying to all steps.
Only meant for experimentation. Built on top of `fe-param-search` branch of Matthias.  

closes #1740 

## Checklist before asking for review

-   [ ] I have performed a self-review of my code
-   [ ] My changes comply with basic sanity checks:
      - I have fixed formatting issues with `./scripts/actions.sh lint`
      - I have run unit tests with `./scripts/actions.sh unit-test`
      - I have documented my code and I have updated the docstrings.
      - I have added unit tests, if relevant
-   [ ] I have tried my changes with data and code:
      - I have run the integration tests with `./scripts/actions.sh integration-test`
      - (bigger changes) I have run a full training and I have written in the comment the run_id(s): `launch-slurm.py --time 60`
      - (bigger changes and experiments) I have shared a hegdedoc in the github issue with all the configurations and runs for this experiments
-   [ ] I have informed and aligned with people impacted by my change:
    - for config changes: the MatterMost channels and/or a design doc
    - for changes of dependencies: the MatterMost software development channel

### CODE REVIEW

**Comment by ankitpatnala** [comment]
cc @SavvasMel
